pkg/codegen/nodejs/gen.go

```diff
@@ -1086,6 +1105,48 @@ func (mod *modContext) genNamespace(w io.Writer, ns *namespace, input bool, leve
 	}
 }
 
+func makeSafeEnumName(name string) string {
+	return makeValidIdentifier(title(name))
+}
+
+func (mod *modContext) genEnum(w io.Writer, enum *schema.EnumType) {
+	enumName := tokenToName(enum.Token)
+	for _, e := range enum.Elements {
+		if e.Name == "" {
+			e.Name = fmt.Sprintf("%v", e.Value)
+		}
+		e.Name = makeSafeEnumName(e.Name)
+		if e.Comment != "" || e.DeprecationMessage != "" {
+			printComment(w, e.Comment, e.DeprecationMessage, "")
+		}
+		fmt.Fprintf(w, "export const %[1]s%[2]s: %[2]s = ", e.Name, enumName)
```

Comment:
This naming is not working out with the existing AWS enums. Changing it.

Reply:
RE: Introduce nodejs-specific enum name overrides

If we do https://github.com/pulumi/pulumi/issues/5494, instead of name overrides, can we have way that would allow us to keep the old names? Whether it's specified in the schema, or we just have a some overlay files for AWS that have the existing constants?

Reply:
@justinvp fair point.

> we just have a some overlay files for AWS that have the existing constants?
I think this might work. We could also add a deprecation message to these constants and redirect folks to the "new" enums. 

It would mean that we would have to update both the schema and the overlay files if there are new enum values added until we can remove the older constants with a major version bump.

I'll give this a go in a separate PR.